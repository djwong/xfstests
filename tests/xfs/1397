#! /bin/bash
# FS QA Test No. 1397
#
# Populate a XFS filesystem and fuzz every rmapbt key/pointer field.
# Do not fix the filesystem, to test metadata verifiers.
# Use xfs_scrub to repair the problems.
#
#-----------------------------------------------------------------------
# Copyright (c) 2018 Oracle, Inc.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1303  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 7 15

_cleanup()
{
	cd /
	rm -rf $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/populate
. ./common/fuzzy

# real QA test starts here
_supported_os Linux
_supported_fs xfs
_require_xfs_scratch_rmapbt
_require_scratch_xfs_fuzz_fields

echo "Format and populate"
_scratch_populate_cached nofill > $seqres.full 2>&1

echo "Fuzz rmapbt keyptr"
_scratch_xfs_fuzz_metadata '' 'none' 'agf 0' 'addr rmaproot' >> $seqres.full
echo "Done fuzzing rmapbt keyptr"

# success, all done
status=0
exit
