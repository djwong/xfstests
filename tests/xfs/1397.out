QA output created by 1397
Format and populate
Fuzz rmapbt keyptr
re-repair failed (1) with magic = zeroes.
re-repair failed (1) with magic = ones.
re-repair failed (1) with magic = firstbit.
re-repair failed (1) with magic = middlebit.
re-repair failed (1) with magic = lastbit.
re-repair failed (1) with magic = add.
re-repair failed (1) with magic = sub.
re-repair failed (1) with magic = random.
re-repair failed (1) with level = zeroes.
re-repair failed (1) with level = ones.
re-repair failed (1) with level = firstbit.
re-repair failed (1) with level = middlebit.
re-repair failed (1) with level = lastbit.
re-repair failed (1) with level = add.
re-repair failed (1) with level = sub.
re-repair failed (1) with level = random.
re-repair failed (1) with numrecs = zeroes.
re-repair failed (1) with numrecs = ones.
re-repair failed (1) with numrecs = firstbit.
re-repair failed (1) with numrecs = middlebit.
re-repair failed (1) with numrecs = lastbit.
re-repair failed (1) with numrecs = add.
re-repair failed (1) with numrecs = sub.
re-repair failed (1) with numrecs = random.
re-repair failed (1) with leftsib = zeroes.
re-repair failed (1) with leftsib = firstbit.
re-repair failed (1) with leftsib = middlebit.
re-repair failed (1) with leftsib = lastbit.
re-repair failed (1) with leftsib = sub.
re-repair failed (1) with leftsib = random.
re-repair failed (1) with rightsib = zeroes.
re-repair failed (1) with rightsib = firstbit.
re-repair failed (1) with rightsib = middlebit.
re-repair failed (1) with rightsib = lastbit.
re-repair failed (1) with rightsib = sub.
re-repair failed (1) with rightsib = random.
re-repair failed (1) with bno = zeroes.
re-repair failed (1) with bno = ones.
re-repair failed (1) with bno = firstbit.
re-repair failed (1) with bno = middlebit.
re-repair failed (1) with bno = lastbit.
re-repair failed (1) with bno = add.
re-repair failed (1) with bno = sub.
re-repair failed (1) with bno = random.
re-repair failed (1) with uuid = zeroes.
re-repair failed (1) with uuid = ones.
re-repair failed (1) with uuid = firstbit.
re-repair failed (1) with uuid = middlebit.
re-repair failed (1) with uuid = lastbit.
re-repair failed (1) with uuid = random.
re-repair failed (1) with owner = ones.
re-repair failed (1) with owner = firstbit.
re-repair failed (1) with owner = middlebit.
re-repair failed (1) with owner = lastbit.
re-repair failed (1) with owner = add.
re-repair failed (1) with owner = sub.
re-repair failed (1) with owner = random.
re-repair failed (1) with crc = zeroes.
re-repair failed (1) with crc = ones.
re-repair failed (1) with crc = firstbit.
re-repair failed (1) with crc = middlebit.
re-repair failed (1) with crc = lastbit.
re-repair failed (1) with crc = add.
re-repair failed (1) with crc = sub.
re-repair failed (1) with crc = random.
re-repair failed (1) with ptrs[1] = zeroes.
re-repair failed (1) with ptrs[1] = ones.
re-repair failed (1) with ptrs[1] = firstbit.
re-repair failed (1) with ptrs[1] = middlebit.
re-repair failed (1) with ptrs[1] = lastbit.
re-repair failed (1) with ptrs[1] = add.
re-repair failed (1) with ptrs[1] = sub.
re-repair failed (1) with ptrs[1] = random.
Done fuzzing rmapbt keyptr
